### ISSUE TITLE
Trakt not visible in PLEX Channel section (Synology)

### ISSUE THREAD

**Issue by kazekira** [trakt/Plex-Trakt-Scrobbler#294]
Hey!

So, last night I managed to install PLEX on my Synology DS410, even though it's not on the official list of supported devices.
Everything works fine and smooth as hell and so far I haven't encountered a single issue, right until I was trying to install Trakt.

I followed the installing instructions and installed it via ssh into the plug-ins folder and restarted the Media Server- nothing.
So I went to the FAQ and tried to fix the plug-in permissions, in case they weren't the way they're supposed to be. After restarting the Media Server- nothing.
It just seems like I can't make the Trakt plug-in appear in my Channels section, no matter what.

So, after hours of wasted time and a lot of frustration: what do?

Cheers,
Matt

As required, here are the links to my PLEX's log files:
com.plexapp.plugins.trakttv.log: https://goo.gl/0pxx9C
Plex Media Server.log: https://goo.gl/x4DEkP

**Comment by fuzeman** [comment]
Currently the plugin doesn't include native libraries for Synology systems running PowerPC processors. Building the required native libraries for PowerPC should be possible, but will require some changes to my build server to get it setup correctly.

I might have some time to work on this later this month, so will update this issue when there is something ready to test.

--------------------------------------

In order to correctly identify this system in the plugin, could you post the output of these commands:
```
cat /proc/cpuinfo
```

```
uname -a
```

**Comment by kazekira** [comment]
Thanks for your fast reply!
Alright, I'm looking forward to seeing you do your magic. :)

Here's the output you requested:

`processor	: 0
cpu		: e500v2
clock		: 1066.560000MHz
revision	: 2.2 (pvr 8021 0022)
bogomips	: 132.94
timebase	: 66474833
platform	: MPC8544 DS
model		: MPC8544DS
Vendor		: Freescale Semiconductor
PVR		: 0x80210022
SVR		: 0x803c0011
PLL setting	: 0x4
Memory		: 512 MB
Memory		: 512 MB`
`Linux kazekira_NAS 2.6.32.12 #5644 Wed Jan 20 22:40:32 CST 2016 ppc GNU/Linux synology_ppc853x_410`

**Comment by padanfain2000** [comment]
Hi Fuzeman,

I am very interested in this as well so here are my specs and Logs. This is from a DS413. Thanks a bunch and keep up this awesome work!


processor       : 0
cpu             : e500v2
clock           : 1066.666000MHz
revision        : 5.1 (pvr 8021 1151)
bogomips        : 133.33

processor       : 1
cpu             : e500v2
clock           : 1066.666000MHz
revision        : 5.1 (pvr 8021 1151)
bogomips        : 133.33

total bogomips  : 266.66
timebase        : 66666667
platform        : P1022 DS
model           : fsl,P1022
Memory          : 1024 MB

Linux DiskStation 2.6.32.12 #5644 SMP Wed Jan 20 22:40:33 CST 2016 ppc GNU/Linux synology_qoriq_413

[Logs.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/205862/Logs.zip)

**Comment by fuzeman** [comment]
Finally have a release for PowerPC ready to test, available here: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/294.zip

**Comment by padanfain2000** [comment]
Great Job Fuzeman,

the channel is now showing up and scrobbling also works. Here are my logs.
[Logs.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/207948/Logs.zip)

Also the folder is not removed anymore and I guess I know why it has been removed in the first place. It's actually rather embarassing. I had copied the trakttv.bundle to the plug-ins-cecxxx folder in @appstore... :/

After I deleted that the correct trakttv.bundle stayed where it was.
Thnx a billion.

**Comment by kazekira** [comment]
Hey,

First of all, thanks a lot for your initiative!

Now, I actually feel kinda dumb. No matter what I tried, trakt keeps being a no-show in the Plex Channels section.

Just for the record: Is it correct to use [Hey,

First of all, thanks a lot for your initiative!

Now, I actually feel kinda dumb. No matter what I tried, trakt keeps being a no-show in the Plex Channels section.

Just for the record: Is it correct to use <url=https://github.com/trakt/Plex-Trakt-Scrobbler/wiki/Installation%20-%20Linux%20(Shell)>this</url> 'installation guide' and just `cp -r Plex-Trakt-Scrobbler-*/Trakttv.bundle "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/"` the Trakttv.bundle which I earlier `wget https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/294.zip -0 Plex-Trakt-Scrobbler.zip` and then unzipped? Because that's exactly what I did. And yea. Nothing happens, still. :(

Here's dem logs: <url=https://goo.gl/rzP4nU>trakttv.log</url>, <url=https://goo.gl/sC58Z9>Media Server.log</url>

I really hope you can help me out somehow.

Thanks again and cheers,
Matt

**Comment by padanfain2000** [comment]
Hi Matt,

I did it using WinSCP. Downloaded Fuzeman's file from above link and extracted it on a windows machine, then stopped Plex, transferred trakttv.bundle to /volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/ and changed (also with WinSCP) owner and group to users:plex. Upon restarting Plex I immediately saw the new traktv channel.

Cheers

pf2k

**Comment by kazekira** [comment]
Hey pf2k,

Thanks for your in-depth explanation of how you managed to make it work for you.
Unfortunately, I'm working on a Macbook, therefore OS X El Capitan, so there's no WinSCP for me.
I tried accessing my NAS with Cyberduck via SFTP (since in Cyberduck, SFTP = SSH), but without luck. There, I was confronted with a bunch of error messages concerning connectivity, which, to be completely honest, I didn't bother to look into any further, because I was just annoyed with everything.

Since I'm still able to access my NAS through the Terminal via SSH, I did all that the things you described via the command shell, also double-checking set owner(s) and user(s).

So basically, I have no clue as to where I could've made a mistake, as the installation is pretty much straight-forward and kind of idiot-proof-ly explained in the installation guide.

Cheers,
Matt

**Comment by padanfain2000** [comment]
You're right it's pretty straightforward and I don't see how you could have done it wrong :( Hopefully Fuzeman can help you out after he has looked through our log files.

I'll keep watching this thread in case I can help in any way.

BR

Oli

**Comment by kazekira** [comment]
Well, thanks anyway for trying. :)
Really looking forward to hearing from Fuzeman again- praise him!

**Comment by fuzeman** [comment]
@padanfain2000 Ah, that would explain why it was deleting the plugin. :smile: Glad it's working now.

----------------

@kazekira Looks like the plugin is still detecting your system as i386 (not PowerPC), probably an issue with Python (or how it's been built).

Just implemented a workaround in the plugin, so do following steps to override architecture detection:

1. Update the plugin with this version: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/294.zip
2. Create the ```Plug-in Support\Data\com.plexapp.plugins.trakttv\advanced.ini``` file with the following contents:

    ```
    [libraries]
    cpu_architecture = PowerPC
    ```

3. Restart Plex Media Server

**Comment by kazekira** [comment]
@fuzeman Thanks for your input.
I tried following everything exactly as you stated, but unfortunately, trakt is still a no-show. There's been an update to the Plex Media Server itself that I installed in the meantime and I also tried reinstalling the trakt plug-in itself, also deleting the `Plug-in Support\Data\com.plexapp.plugins.trakttv\` folder beforehand.
After reinstalling the plug-in and inserting the .ini file, nothing happens. Still the same. :(

Here's (again) the current logs:
[com.plexapp.plugins.trakttv.log](https://drive.google.com/a/radakovits.net/file/d/0B1Z2tp8ZcoNRUGMxcU1UTWtNSDQ/view?usp=drivesdk)
[Plex Media Server.log] (https://drive.google.com/a/radakovits.net/file/d/0B1Z2tp8ZcoNRYy1JRmp1bUFXMG8/view?usp=drivesdk)

**Comment by fuzeman** [comment]
@kazekira Based on the log file, it looks like you are still using an older version of the plugin.

Did you install the latest changes (from https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/294.zip)?

**Comment by kazekira** [comment]
@fuzeman Yup, I did. Just to be completely sure, I just wiped everything clean from old files and directories within the `/root` directory and did a complete reinstall of the plugin, using the exact link you just sent. But now, the plugin not only is a no-show, it seems that it doesn't even attempt to start, or at least there were no logs generated during the last couple of restarts.
So, the only log I'm able to submit now is the current [Plex Media Server.log](https://drive.google.com/a/radakovits.net/file/d/0B1Z2tp8ZcoNRX1pfQ0hBaW1kWkU/view?usp=drivesdk).
Honestly, I'm kind of desperate now and almost ready to throw the towel about the whole thing. :/

**Comment by fuzeman** [comment]
@kazekira From that log file:
```
ERROR - Error reading file [/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Info.plist].
ERROR - Unable to scan bundle /volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle: Empty or non-existant file.
```

----------------------

I'm guessing it's either caused by:
 - Not restarting Plex Media Server after plugin re-installation - PMS can sometimes start plugins before they have finished being copied into the "Plug-ins" directory, restarting the server resolves this issue.
 - Incorrect permissions, you might need to run:

    ```
    sudo chown -R plex:plex "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle"
    sudo chmod -R 770 "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle"
    ```

    If the ```chown``` command fails, your system might be using a different user + group for plex, you can find the correct owner with:
    ```
    stat --format '%U:%G' "/volume1/Plex/Library/Application Support/Plex Media Server"
    ```
    then use the result from that command instead of ```plex:plex```

**Comment by kazekira** [comment]
Oh Christ.
I can't believe that in the end it all came down to me messing up the permissions.
Reapplying user + group and `chmod -R 770 "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle"` finally did the trick and now trakt is actually showing up and syncing right now!
I really hope that now that the hard part's over, everything starts to work smoothly. If I encounter any further issues with the plug-in itself, I'll send another cry for help.

Anyways, thanks A LOT @fuzeman for your continuous help. I'd've possibly already thrown my NAS out of the windows if it weren't for you.

**Comment by fuzeman** [comment]
@kazekira No worries, glad you got it working :smile:

**Comment by fuzeman** [comment]
Changes released in [v1.0.2.0](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.2.0)
